sdk/translation/ai-translation-text/src/customClient.ts

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { getClient, ClientOptions } from "@azure-rest/core-client";
+import * as coreRestPipeline from "@azure/core-rest-pipeline";
+import {
+  PipelineRequest,
+  PipelineResponse,
+  SendRequest
+} from "@azure/core-rest-pipeline";
+import { TextTranslationClient } from "./generated/clientDefinitions";
+import {
+  TranslatorCredential,
+  TranslatorAuthenticationPolicy,
+  TranslatorAzureKeyAuthenticationPolicy
+} from "./authentication";
+import {
+  AzureKeyCredential,
+  TokenCredential
+} from "@azure/core-auth";
+
+const DEFAULT_SCOPE = "https://cognitiveservices.azure.com/.default";
+const DEFAULT_ENPOINT = "https://api.cognitive.microsofttranslator.com";
+const PLATFORM_HOST = "cognitiveservices";
+const PLATFORM_PATH = "/translator/text/v3.0";
+
+/**
+ * Initialize a new instance of `TextTranslationClient`
+ * @param endpoint type: string, Supported Text Translation endpoints (protocol and hostname, for example:
+ *     https://api.cognitive.microsofttranslator.com).
+ * @param options type: ClientOptions, the parameter for all optional parameters
+ */
+export default function createClient(
+  endpoint: undefined | string,
+  credential: undefined | TranslatorCredential | AzureKeyCredential | TokenCredential = undefined,
+  options: ClientOptions = {}
+): TextTranslationClient {
+  let serviceEndpoint: string;
+  if (!endpoint) {
+    serviceEndpoint = DEFAULT_ENPOINT;
+  } else if (endpoint.toLowerCase().indexOf(PLATFORM_HOST) != -1) {
+    serviceEndpoint = `${endpoint}${PLATFORM_PATH}`;
```

Comment:
Is `PLATFORM_PATH` not specified in the swagger/typespec? it feels weird to be needed to be added here. Also, does it need to be added to the global endpoint too?

Reply:
We have a bit special situations with endpoints. I am doing the same handling in other languages. Briefly "why":
- translator supports translator's endpoints (https://learn.microsoft.com/en-us/azure/cognitive-services/translator/reference/v3-0-reference) - e.g. `https://api.cognitive.microsofttranslator.com/translate` for translate
- translator also supports "APIM" which is endpoint managed by cognitive services platform. This is used when customers want to use some advanced features like VNET. In that case, the host AND path is different, so the translate operation full path is: `https://mimat-canada.cognitiveservices.azure.com/translator/text/v3.0/translate` (`mimat-canada` is name of the CS resource).

For the second question - no, global endpoint's path is just `/translate`.